MonoGame.Framework/Input/Keyboard.Windows.cs

```diff
@@ -4,21 +4,57 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.Xna.Framework.Input
 {
     public static partial class Keyboard
     {
-        static List<Keys> _keys;
+        private static readonly Keys[] DefinedKeys;
+
+        private static readonly byte[] _keyState = new byte[256];
+        private static readonly List<Keys> _keys = new List<Keys>(10);
+
+        private static bool _isActive;
+
+        [DllImport("user32.dll")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool GetKeyboardState(byte[] lpKeyState);
+
+        private static readonly Predicate<Keys> IsKeyReleasedPredicate = key => !IsKeyPressed(key);
+
+        static Keyboard()
+        {
+            DefinedKeys = Enum.GetValues(typeof(Keys)).Cast<Keys>()
+                .Where(k => ((int)k >= 1) && ((int)k <= 255))
+                .ToArray();
```

Comment:
Yeah, I know it can be bad from a memory pressure point of view. But I figured it'd be ok since it's only called once when the static constructor is first called. But I can rewrite it without using LINQ if you prefer.